packages/synapse-react-client/src/lib/containers/oauth/CreateOAuthClient.tsx

```diff
@@ -64,17 +93,21 @@ export const CreateOAuthModal: React.FunctionComponent<
     setTosUri(client?.tos_uri ?? '')
   }, [isShowingModal, client])
 
-  useEffect(() => {
-    const stateArr = redirectUris?.map(str => str.uri)
-    const propArr = client?.redirect_uris
-
-    client &&
-    (stateArr?.length !== propArr?.length ||
-      !stateArr?.every(el => propArr?.includes(el)) ||
-      sectorUri != client?.sector_identifier_uri)
-      ? setWarnTrigger(true)
-      : setWarnTrigger(false)
-  }, [redirectUris, sectorUri, client])
+  useDebouncedEffect(
+    () => {
+      if (accessToken) {
+        // SWC-6365: use the pre-check service to determine if we need to show a warning on edit
+        SynapseClient.isOAuthClientReverificationRequired(
+          getOAuthClient(),
```

Comment:
Following above suggestion
```suggestion
          oAuthClient,
```